pkg/operation/botanist/component/logging/fluentoperator/fluent_operator.go

```diff
@@ -0,0 +1,501 @@
+	for _, clusterFilter := range clusterFilters {
+		operatorCustomResources = append(operatorCustomResources, clusterFilter)
+	}
+
+	clusterParsers := customresources.GetClusterParsers(f.getCustomResourcesLabels())
+	for _, clusterParser := range clusterParsers {
+		operatorCustomResources = append(operatorCustomResources, clusterParser)
+	}
+
+	clusterOutputs := customresources.GetClusterOutputs(f.getCustomResourcesLabels())
+	for _, clusterOutput := range clusterOutputs {
+		operatorCustomResources = append(operatorCustomResources, clusterOutput)
+	}
+
+	for _, clusterInput := range f.additionalInputs {
+		operatorCustomResources = append(operatorCustomResources, clusterInput)
+	}
+
+	for _, clusterFilter := range f.additionalFilters {
+		operatorCustomResources = append(operatorCustomResources, clusterFilter)
+	}
+
+	for _, clusterParser := range f.additionalParsers {
+		operatorCustomResources = append(operatorCustomResources, clusterParser)
+	}
+
+	serializedOperatorResources, err := operatorRegistry.AddAllAndSerialize(operatorResources...)
+	if err != nil {
+		return err
+	}
+
+	serializedOperatorCustomResources, err := operatorCustomResourcesRegistry.AddAllAndSerialize(operatorCustomResources...)
+	if err != nil {
+		return err
+	}
+
+	if err := managedresources.CreateForSeed(ctx, f.client, f.namespace, CustomResourcesManagedResourceName, false, serializedOperatorCustomResources); err != nil {
+		return err
+	}
+
+	return managedresources.CreateForSeed(ctx, f.client, f.namespace, OperatorManagedResourceName, false, serializedOperatorResources)
+}
+
+func (f *fluentOperator) Destroy(ctx context.Context) error {
+	if err := managedresources.DeleteForSeed(ctx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.DeleteForSeed(ctx, f.client, f.namespace, OperatorManagedResourceName)
+}
+
+// TimeoutWaitForManagedResources is the timeout used while waiting the managed resources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResources = 5 * time.Minute
+
+func (f *fluentOperator) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResources)
+	defer cancel()
+
+	if err := managedresources.WaitUntilHealthy(timeoutCtx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, f.client, f.namespace, OperatorManagedResourceName)
+}
+
+func (f *fluentOperator) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResources)
+	defer cancel()
+
+	if err := managedresources.WaitUntilDeleted(timeoutCtx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, f.client, f.namespace, OperatorManagedResourceName)
```

Comment:
Probably we should wait in parallel

Reply:
Nope because if the FB operator is deleted before it deletes the FB Daemonset (which happens when the FluentBit CustomResoruce is deleted) the daemonset will leak.

Reply:
I don't understand. We wait here until the `ManagedResource`s are deleted - there is no code in between. Why can't we wait in parallel? The explanation doesn't make sense to me.
Deletion of the backed resources is instrumented by `gardener-resource-manager` anyways.

Reply:
You mean to delete the resources in the MR parallel or to delete both managed MRs parallel?

Reply:
I actually do not understand what means to wait in parallel. What should be parallel?
The constraint here is that I want to be sure that the first MR is deleted entirely before starting the deletion of the second one.

Reply:
> The constraint here is that I want to be sure that the first MR is deleted entirely before starting the deletion of the second one.

I don't see this being implemented. `Destroy` deletes both `ManagedResource`s, and `WaitCleanup` first waits for the first `ManagedResource` and only when it's gone then it waits for the second one. Hence, I proposed to wait in parallel instead of waiting sequentially.

I don't think that what you try to achieve is possible with just one component.